libs/core/src/components/pds-input/pds-input.tsx

```diff
@@ -137,6 +152,89 @@ export class PdsInput {
    */
   @State() hasFocus = false;
 
+  private updateAddonWidths() {
+    requestAnimationFrame(() => {
+      if (this.prefixEl) {
+        const prefixWidth = this.prefixEl.offsetWidth;
+        this.el.style.setProperty('--prefix-width', `${prefixWidth}px`);
+      }
+
+      if (this.suffixEl) {
+        const suffixWidth = this.suffixEl.offsetWidth;
+        this.el.style.setProperty('--suffix-width', `${suffixWidth}px`);
+      }
+    });
+  }
+
+  private renderPrefix() {
+    const hasPrefix = this.el.querySelector('[slot="prefix"]') !== null;
+    if (hasPrefix) {
+      return (
+        <div class="pds-input__prefix" part="prefix" ref={(el) => this.prefixEl = el as HTMLElement}>
+          <slot name="prefix" onSlotchange={() => this.updateAddonWidths()}></slot>
+        </div>
+      );
+    }
+    return null;
+  }
+
+  private renderSuffix() {
+    const hasSuffix = this.el.querySelector('[slot="suffix"]') !== null;
+    if (hasSuffix) {
+      return (
+        <div class="pds-input__suffix" part="suffix" ref={(el) => this.suffixEl = el as HTMLElement}>
+          <slot name="suffix" onSlotchange={() => this.updateAddonWidths()}></slot>
+        </div>
+      );
+    }
+    return null;
+  }
+
+  private renderPrepend() {
+    const hasPrepend = this.el.querySelector('[slot="prepend"]') !== null;
+    if (hasPrepend) {
+      return (
+        <div class="pds-input__prepend" part="prepend">
+          <slot name="prepend"></slot>
+        </div>
+      );
+    }
+    return null;
+  }
+
+  private renderAppend() {
+    const hasAppend = this.el.querySelector('[slot="append"]') !== null;
+    if (hasAppend) {
+      return (
+        <div class="pds-input__append" part="append">
+          <slot name="append"></slot>
+        </div>
+      );
+    }
+    return null;
+  }
+
+  componentWillLoad() {
+    this.inheritedAttributes = {
+      ...inheritAriaAttributes(this.el)
+    };
+    this.hasPrefix = this.el.querySelector('[slot="prefix"]') !== null;
+    this.hasSuffix = this.el.querySelector('[slot="suffix"]') !== null;
+    this.hasPrepend = this.el.querySelector('[slot="prepend"]') !== null;
+    this.hasAppend = this.el.querySelector('[slot="append"]') !== null;
```

Comment:
Can you elaborate more on these (e.g, the difference between prefix vs prepend and suffix and append)?

Reply:
Updated the docs to call out that prefix/suffix items appear visually within the input, while append/prepend appear next to it